### PR TITLE
[WiP] Send log of failed npm to slack

### PR DESCRIPTION
**Pull request by yurynix**
TL;DR - Sometimes `npm install` fails, we'd like to collect the logs from those failures.

Some projects, like ecom migrated to `@wix` scoped packages and pulled them in the CI from `registry.npmjs.org`.
Ecom said that makes their build unstable, so we went and did some investigation.
It turns out, fetching packages/metadata from `registry.npmjs.org` sucks, it manifests itself with 2 “main” errors:
1.
```
npm ERR! code EINTEGRITY
npm ERR! sha512-8VrRnZmtbtHcg7oBlnaaqlddjnhBVHwg/igG0v3ppGGgrgtwMcKYPDhvhS+koVVKR44hz5D2FxL5nLhKo51/PQ== integrity checksum failed when using sha512: wanted sha512-8VrRnZmtbtHcg7oBlnaaqlddjnhBVHwg/igG0v3ppGGgrgtwMcKYPDhvhS+koVVKR44hz5D2FxL5nLhKo51/PQ== but got sha512-QUOU4WiDU2qX4UzPDi50gcPhYWcRBLCaVN1P6lZfYDz2HYg5diNOSvPth9HXqKRO3Z5C5jNR4D/VMbKFCLADEg==. (1973960 bytes)
```

(this is not to be confused with: https://wix.slack.com/archives/C0NJPS8N5/p1550650355083300?thread_ts=1550602663.079600&cid=C0NJPS8N5)

2.
```
npm ERR! Unexpected end of input at 1:2135631
npm ERR! ,“grunt-cli”:“^0.1.13”,“grunt-includes”:“0.5.4”,“grunt-webpack”:“~3.0
npm ERR!
```

example: http://tc.dev.wixpress.com/viewLog.html?buildId=80560936&buildTypeId=WixEcommerceServices_WixEcommerceStoreManager&tab=buildLog&state=&expand=all&_focus=520#_state=

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -49,3 +51,106 @@ export function readJsonFile(name) {
...
+      reject(err);
+    });
+    gzipStream.on('end', () => {
+      resolve(Buffer.concat(buffers));
+    });
+  });
+}
+
+function sendFileToSlack(path, title, filename) {
+  if (!process.env.NPM_CI_SLACK_TOKEN) {
```
Comment (yairhaimo) [inline src/utils.js]: will this break the build or fail gracefully?

File: src/build.js [hunk 2]
```diff
@@ -2,12 +2,12 @@ import {execCommand} from './utils';
...
 
-export function build(buildType) {
+export async function build(buildType) {
   if (process.env.APPLITOOLS_GITHUB_FT) {
     setApplitoolsId();
   }
 
-  install();
+  await install();
   execCommand('npm run build --if-present');
```
Comment (yairhaimo) [inline src/build.js]: do we want to `await` on this?

File: src/utils.js (hunk 1 above)
Comment (yurynix) [reply src/utils.js]: This will just print an error but continue, it's wrapper in a `try {} catch` https://github.com/wix/npm-ci-scripts/pull/17/files/64fe5ccabcc4644206024093f1caefe64046d11f#diff-2b4ca49d4bb0a774c4d4c1672d7aa781R127

File: src/build.js (hunk 2 above)
Comment (yurynix) [reply src/build.js]: That's the old sync command, so it's already sync

**Comment by yurynix** [comment]
Merged that to test on CI, might revert.

**Comment by yurynix** [comment]
Looks fine on CI.
